modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/SidebarPanel.js

```diff
@@ -14,26 +14,86 @@
 
 import ClayAlert from '@clayui/alert';
 import ClayLoadingIndicator from '@clayui/loading-indicator';
-import {useTimeout} from 'frontend-js-react-web';
+import {useIsMounted, useTimeout} from 'frontend-js-react-web';
 import {fetch} from 'frontend-js-web';
-import React, {useEffect, useImperativeHandle, useRef, useState} from 'react';
+import React, {useEffect, useImperativeHandle, useReducer, useRef} from 'react';
 
 import Sidebar from './components/Sidebar';
 
+const initialState = {
+	data: null,
+	error: null,
+	html: null,
+	loading: false,
+	open: true,
+};
+
 const SidebarPanel = React.forwardRef(
 	({fetchURL, onClose, viewComponent: View}, ref) => {
 		const CurrentView = useRef(View);
 
-		const [error, setError] = useState();
-		const [isLoading, setIsLoading] = useState();
-		const [isOpen, setIsOpen] = useState(true);
-		const [resourceData, setResourceData] = useState();
-
 		const delay = useTimeout();
 
+		const isMounted = useIsMounted();
+
+		const dataReducer = (state, action) => {
+			if (isMounted()) {
```

Comment:
Hoist the `isMounted()` call outside of the reducer, and instead put it around/before the `dispatch` calls. Reason: reducer should be a pure function (in fact, you should be able to move it outside of the component), and I am not sure this will have the desired effect here (ie. if you dispatch an action when the component it unmounted, it will probably warn/error even though you return `undefined` here).